forest/cli/src/cli/mod.rs

```diff
@@ -19,84 +19,73 @@ mod state_cmd;
 mod sync_cmd;
 mod wallet_cmd;
 
-pub(super) use self::auth_cmd::AuthCommands;
-pub(super) use self::chain_cmd::ChainCommands;
-pub(super) use self::db_cmd::DBCommands;
 pub(super) use self::fetch_params_cmd::FetchCommands;
-pub(super) use self::mpool_cmd::MpoolCommands;
-pub(super) use self::net_cmd::NetCommands;
 pub(super) use self::send_cmd::SendCommand;
-pub(super) use self::snapshot_cmd::SnapshotCommands;
-pub(super) use self::state_cmd::StateCommands;
-pub(super) use self::sync_cmd::SyncCommands;
-pub(super) use self::wallet_cmd::WalletCommands;
-pub(crate) use forest_cli_shared::cli::{Config, FOREST_VERSION_STRING};
+use self::{
+    auth_cmd::AuthCommandsStruct, chain_cmd::ChainCommandsStruct, config_cmd::ConfigCommandsStruct,
+    db_cmd::DBCommandsStruct, mpool_cmd::MpoolCommandsStruct, net_cmd::NetCommandsStruct,
+    snapshot_cmd::SnapshotCommandsStruct, state_cmd::StateCommandsStruct,
+    sync_cmd::SyncCommandsStruct, wallet_cmd::WalletCommandsStruct,
+};
+use clap::Parser;
+pub(crate) use forest_cli_shared::cli::Config;
 
-use crate::cli::config_cmd::ConfigCommands;
 use cid::Cid;
 use forest_blocks::tipset_json::TipsetJson;
 use forest_cli_shared::cli::CliOpts;
 use jsonrpc_v2::Error as JsonRpcError;
 use log::error;
 use serde::Serialize;
 use std::io::{self, Write};
-use structopt::StructOpt;
 
 /// CLI structure generated when interacting with Forest binary
-#[derive(StructOpt)]
-#[structopt(
-    name = env!("CARGO_PKG_NAME"),
-    version = FOREST_VERSION_STRING.as_str(),
-    about = env!("CARGO_PKG_DESCRIPTION"),
-    author = env!("CARGO_PKG_AUTHORS")
-)]
+#[derive(Parser)]
 pub struct Cli {
-    #[structopt(flatten)]
+    #[command(flatten)]
     pub opts: CliOpts,
-    #[structopt(subcommand)]
+    #[command(subcommand)]
     pub cmd: Subcommand,
 }
 
 /// Forest binary sub-commands available.
-#[derive(StructOpt)]
-#[structopt(setting = structopt::clap::AppSettings::VersionlessSubcommands)]
+#[derive(clap::Subcommand)]
 pub enum Subcommand {
     /// Download parameters for generating and verifying proofs for given size
-    #[structopt(name = "fetch-params")]
+    #[command(name = "fetch-params")]
     Fetch(FetchCommands),
 
     /// Interact with Filecoin blockchain
-    Chain(ChainCommands),
+    Chain(ChainCommandsStruct),
```

Comment:
Why use inconsistent pattern here between `Fetch` and the others? Any chance to remove the `*Struct` boilerplate code?

Reply:
Unfortunately, in clap you cannot have subcommands that are enums of subcommands, therefore this workaround. That's the main difference between  `ChainCommands` and `Fetch`

Reply:
Thanks!

Reply:
@tyshko5 Why do you say you cannot have them? https://gist.github.com/LesnyRumcajs/dcd8be1f7bdcefcc438e2cdfa2da4a36

Reply:
Had an issue with this. Fixed it